app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt

```diff
@@ -26,17 +27,17 @@ class HomeRepoImpl @Inject constructor(
     override fun fetchSurveyList(
         pageNumber: Int,
         pageSize: Int,
-        getNumberOfPage: (totalPage: Int) -> Unit
-    ): Flow<ResourceState<Unit>> = flow {
+        isClearCache: Boolean
+    ): Flow<ResourceState<Meta>> = flow {
         emit(ResourceState.Loading)
         val apiResult = safeApiCall(Dispatchers.IO) { apiInterface.getSurveyList(pageNumber, pageSize) }
         when (apiResult) {
             is ResourceState.Success -> {
                 apiResult.data.data?.let { surveyResponseList ->
+                    if (isClearCache) surveyDao.clearSurveys()
                     surveyDao.addSurveys(surveyResponseList.map { it.toSurvey() })
-                    getNumberOfPage(apiResult.data.meta?.pages ?: 0)
                 }
-                emit(ResourceState.Success(Unit))
+                emit(ResourceState.Success(apiResult.data.meta?.toMeta() ?: Meta()))
```

Comment:
and then this will become 
```suggestion
                emit(ResourceState.Success(apiResult.data.meta.toMeta()))
```
What do you think? 😄